### DISCUSSION TITLE
Undo the hardcoding of the outpost deathmatch mission

### DISCUSSION THREAD

**Discussion by profole** [FakeFishGames/Barotrauma#15417] [Unstable feedback]
In current stable the xml for the outpost deathmatch mission is
```xml
  <CombatMission
    identifier="outpostdeathmatch"
    type="OutpostCombat"
    multiplayeronly="true"
    commonness="100"
    reward="2000"
    pointsperkill="10"
    allowrespawning="true"
    wincondition="KillCount"
    loadsubmarines="false"
    AllowOutpostNPCs="False"
    ForceOutpostGenerationParameters="colonypvp"
    AllowOutpostSelectionFromTag="PvPOutpost"
    ForceRespawnMode="MidRound">
    <LocationType identifier="city" />
    <Icon texture="Content/UI/MissionIcons2.png" sourcerect="0,768,256,256" color="243,146,55"/>
    <message text="missionmessage0.outpostdeathmatch"/>
    <message text="missionmessage1.outpostdeathmatch"/>
    <TriggerEvent state="0" delay="10" eventtag="deathmatchweapondrop" />
    <OverrideMusic state="0" type="pvp" />
  </CombatMission>
```
Which was changed to the following one in unstable
```xml
  <CombatMission
    identifier="outpostdeathmatch"
    type="OutpostCombat"
    multiplayeronly="true"
    commonness="100"
    reward="2000"
    pointsperkill="10"
    allowrespawning="true"
    wincondition="KillCount"
    loadsubmarines="false"
    AllowOutpostNPCs="False"
    ForceOutpostGenerationParameters="colonypvp"
    AllowOutpostSelectionFromTag="PvPOutpost"
    ForceRespawnMode="MidRound">
    <LocationType identifier="city" />
    <Icon texture="Content/UI/MissionIcons2.png" sourcerect="0,768,256,256" color="243,146,55"/>
    <message text="missionmessage0.outpostdeathmatch"/>
    <message text="missionmessage1.outpostdeathmatch"/>
    <OverrideMusic state="0" type="pvp" />
  </CombatMission>
```

Notably, the `<TriggerEvent state="0" delay="10" eventtag="deathmatchweapondrop" />` line was removed and its most likely hardcoded into the mission/chosen map now.
I assume this is done to get around making xml stuff to check which map is being played (since the old one couldnt do that without making a new missiontype and we now need that since ruin map needs to spawn different things), but this just cant be it.

My mod "Better Outpost Deathmatch Framework" for example, used the old system to add a new custom event triggered by a custom missiontype to allow people to still play PvP vanilla while having my mod enabled.

Considering that the goal is to de-hardcode more stuff for xml modding, this shouldnt be done.
I suggest adding a new xml attribute for TriggerEvent to check against the selected map (something like "OutpostTag" or "MapTag") instead.

**Comment by Regalis11** [comment]
The event is defined in the outpost file now, it's not hard-coded. I would argue that this makes the system a lot more flexible: checking the map in the event would get really messy and cause conflicts between mods, I think it's a lot better that the creator of the outpost/map is free to choose what kind of an event is used for the weapon drops.

**Comment by JeroonBosch** [comment]
> My mod "Better Outpost Deathmatch Framework" for example, used the old system to add a new custom event triggered by a custom missiontype to allow people to still play PvP vanilla while having my mod enabled.

Re-adding `<TriggerEvent state="0" delay="10" eventtag="deathmatchweapondrop" />` to your mod's XML should still work though, and if that's undesirable, you can add custom copies of the PvP maps with the events you want.

Is there any use case that was previously possible, but isn't currently? As far as we know, we didn't remove or hardcode any functionality?

**Reply by profole** [reply]
I originally missed that you do it in editor now (iirc it wasnt actually listed in unstable patchnotes).
Other than that, if this didn't remove the option to trigger events using missions, I don't think any functionality was removed.
Feel free to close!

**Reply by Regalis11** [reply]
> I originally missed that you do it in editor now (iirc it wasnt actually listed in unstable patchnotes).

Yeah it seems we forgot to mention this in the patch notes - it's good that this came up, I think others would've been confused about this too. We've added a note about this now, coming up in the next unstable build. :)
